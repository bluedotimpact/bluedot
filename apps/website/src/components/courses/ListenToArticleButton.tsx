import React from 'react';

type ListenToArticleButtonProps = {
  audioUrl: string;
  resourceTitle: string;
};

const ListenToArticleButton: React.FC<ListenToArticleButtonProps> = ({ audioUrl, resourceTitle }) => {
  const handleClick = () => {
    // Open Spotify URL in new tab
    window.open(audioUrl, '_blank', 'noopener,noreferrer');
  };

  return (
    <button
      onClick={handleClick}
      className="flex flex-row items-center gap-1.5 h-[18px] group cursor-pointer transition-colors duration-200 bg-transparent border-none p-0"
      aria-label={`Listen to article: ${resourceTitle} (opens in Spotify)`}
      type="button"
    >
      {/* Play icon */}
      <svg
        width="16"
        height="16"
        viewBox="0 0 16 16"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
        aria-hidden="true"
      >
        <rect
          x="0.625"
          y="0.625"
          width="14.75"
          height="14.75"
          rx="7.375"
          stroke="#6A6F7A"
          strokeWidth="1.25"
          className="group-hover:stroke-[#13132E] transition-colors duration-200"
        />
        <path
          d="M6 10.4839V5.51675C6 5.13551 6.40956 4.89452 6.74282 5.07967L11.2133 7.56325C11.5562 7.75375 11.5562 8.2469 11.2133 8.4374L6.74282 10.921C6.40956 11.1061 6 10.8651 6 10.4839Z"
          fill="#6A6F7A"
          className="group-hover:fill-[#13132E] transition-colors duration-200"
        />
      </svg>

      {/* Text */}
      <span className="text-[13px] font-medium leading-[140%] tracking-[-0.005em] text-[#6A6F7A] group-hover:text-[#13132E] transition-colors duration-200">
        Listen to article
      </span>
    </button>
  );
};

export default ListenToArticleButton;
