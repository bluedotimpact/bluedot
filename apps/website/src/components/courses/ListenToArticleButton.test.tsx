import { render } from '@testing-library/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import userEvent from '@testing-library/user-event';
import {
  describe, expect, it, vi,
} from 'vitest';
import ListenToArticleButton from './ListenToArticleButton';

describe('ListenToArticleButton', () => {
  const mockAudioUrl = 'https://open.spotify.com/episode/test123';
  const mockResourceTitle = 'Introduction to AI Safety';

  it('should render correctly', () => {
    const { container } = render(
      <ListenToArticleButton
        audioUrl={mockAudioUrl}
        resourceTitle={mockResourceTitle}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with proper aria-label', () => {
    const { getByRole } = render(
      <ListenToArticleButton
        audioUrl={mockAudioUrl}
        resourceTitle={mockResourceTitle}
      />,
    );
    const button = getByRole('button');
    expect(button.getAttribute('aria-label')).toBe(
      'Listen to article: Introduction to AI Safety (opens in Spotify)',
    );
  });

  it('should open audio URL in new tab when clicked', async () => {
    const user = userEvent.setup();
    const mockOpen = vi.fn();
    const originalOpen = window.open;
    window.open = mockOpen;

    const { getByRole } = render(
      <ListenToArticleButton
        audioUrl={mockAudioUrl}
        resourceTitle={mockResourceTitle}
      />,
    );

    const button = getByRole('button');
    await user.click(button);

    expect(mockOpen).toHaveBeenCalledWith(
      mockAudioUrl,
      '_blank',
      'noopener,noreferrer',
    );

    window.open = originalOpen;
  });

  it('should render with different resource titles', () => {
    const { container, rerender } = render(
      <ListenToArticleButton
        audioUrl={mockAudioUrl}
        resourceTitle="First Article"
      />,
    );
    const firstRender = container.innerHTML;

    rerender(
      <ListenToArticleButton
        audioUrl={mockAudioUrl}
        resourceTitle="Second Article with a Much Longer Title"
      />,
    );
    const secondRender = container.innerHTML;

    expect(firstRender).not.toEqual(secondRender);
  });
});
