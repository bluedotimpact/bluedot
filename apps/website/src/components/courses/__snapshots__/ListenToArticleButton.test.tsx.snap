// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListenToArticleButton > should render correctly 1`] = `
<button
  aria-label="Listen to article: Introduction to AI Safety (opens in Spotify)"
  class="flex flex-row items-center gap-1.5 h-[18px] group cursor-pointer transition-colors duration-200 bg-transparent border-none p-0"
  type="button"
>
  <svg
    aria-hidden="true"
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      class="group-hover:stroke-[#13132E] transition-colors duration-200"
      height="14.75"
      rx="7.375"
      stroke="#6A6F7A"
      stroke-width="1.25"
      width="14.75"
      x="0.625"
      y="0.625"
    />
    <path
      class="group-hover:fill-[#13132E] transition-colors duration-200"
      d="M6 10.4839V5.51675C6 5.13551 6.40956 4.89452 6.74282 5.07967L11.2133 7.56325C11.5562 7.75375 11.5562 8.2469 11.2133 8.4374L6.74282 10.921C6.40956 11.1061 6 10.8651 6 10.4839Z"
      fill="#6A6F7A"
    />
  </svg>
  <span
    class="text-[13px] font-medium leading-[140%] tracking-[-0.005em] text-[#6A6F7A] group-hover:text-[#13132E] transition-colors duration-200"
  >
    Listen to article
  </span>
</button>
`;
